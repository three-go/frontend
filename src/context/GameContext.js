import React, { createContext, useState } from "react";

const GameContext = createContext();

export const GameContextProvider = ({ children }) => {
  const [currentGameKey, setCurrentGameKey] = useState("");

  return (
    <GameContext.Provider value={{ currentGameKey, setCurrentGameKey }}>
      {children}
    </GameContext.Provider>
  );
};

export default GameContext;
