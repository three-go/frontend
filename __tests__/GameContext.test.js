import React, { useContext } from "react";

import { render, fireEvent } from "@testing-library/react-native";
import { Text, Button } from "react-native";

import GameContext, { GameContextProvider } from "../src/context/GameContext";

describe("Context API", () => {
  it("GameContext", () => {
    const testContextValue = "This is context test";

    const TestComponent = () => {
      const { currentGameKey, setCurrentGameKey } = useContext(GameContext);
      return (
        <>
          <Text>{currentGameKey}</Text>
          <Button
            onPress={() => setCurrentGameKey(testContextValue)}
            title="Testing Button"
          />
        </>
      );
    };

    const component = render(
      <GameContextProvider>
        <TestComponent />
      </GameContextProvider>
    );

    expect(component.getByText("Testing Button")).toBeTruthy();

    const testButton = component.getByText("Testing Button");
    fireEvent.press(testButton);

    expect(component.getByText(testContextValue)).toBeTruthy();
    expect(component.toJSON).toMatchSnapshot();
  });
});
