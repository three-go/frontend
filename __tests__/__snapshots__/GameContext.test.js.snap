// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Context API GameContext 1`] = `[Function]`;
